tests/test_backend.py

```diff
@@ -0,0 +1,187 @@
+from http import HTTPStatus
+from unittest import TestCase
+from unittest.mock import patch, MagicMock
+import torch
+from torch.fx import GraphModule
+import hidet
+from hidet.graph.frontend.torch.dynamo_backends import get_flow_graph
+from centml.compiler.backend import Runner
+from centml.compiler.server_compilation import CompilationStatus
+from centml.compiler import config_instance
+
+
+def get_graph_module(name):
+    model = torch.hub.load('pytorch/vision:v0.10.0', name, pretrained=True, verbose=False).eval()
+    graph_module: GraphModule = torch.fx.symbolic_trace(model)
+    interpreter = hidet.frontend.from_torch(graph_module)
+    return graph_module, interpreter
+
+
+class TestGetModelId(TestCase):
+    @patch('threading.Thread.start')
+    def setUp(self, mock_thread) -> None:
+        self.inputs = [torch.zeros(1, 3, 224, 224)]
+        graph_module, interpreter = get_graph_module('resnet18')
+        graph_module_34, interpreter_34 = get_graph_module('resnet34')
+        self.flow_graph, _, _ = get_flow_graph(interpreter, self.inputs)
+        self.flow_graph_34, _, _ = get_flow_graph(interpreter_34, self.inputs)
+        self.runner = Runner(graph_module, None)
+        self.runner_34 = Runner(graph_module_34, None)
+
+    @patch('hidet.save_graph')
+    def test_none_flow_graph(self, mock_save_graph):
+        with self.assertRaises(Exception) as context:
+            self.runner._get_model_id(None)
+
+        self.assertIn("Getting model id: flow graph is None.", str(context.exception))
+        mock_save_graph.assert_not_called()
+
+    @patch('hidet.save_graph')
+    def test_exception_on_save_graph_failure(self, mock_save_graph):
+        mock_save_graph.side_effect = Exception("Test Exception")
+        with self.assertRaises(Exception) as context:
+            self.runner._get_model_id(self.flow_graph)
+
+        self.assertIn("Getting model id: failed to save FlowGraph.", str(context.exception))
+        mock_save_graph.assert_called_once()
+
+    def test_output_consistency(self):
+        model_id1 = self.runner._get_model_id(self.flow_graph)
+        model_id2 = self.runner._get_model_id(self.flow_graph)
+        self.assertEqual(model_id1, model_id2)
```

Comment:
Lets also do a consistency check on getting model id from torch.fx graph.

Reply:
We don't currently use the torch.fx graph to get the model id. We always convert torch.fx --> FlowGraph --> pickled FlowGraph binary --> hash/model_id. 

When I tried using the torch.fx graph in the past the hash was inconsistent. Did you want a test to see if that hash remains inconsistent?

Reply:
Verify that given a model, verify it always:
 - Generates the same torch.fx graph
 - That torch.fx graph always generates the same FlowGraph